cdap-gateway/src/main/java/co/cask/cdap/gateway/router/RouterPathLookup.java

```diff
@@ -155,10 +155,13 @@ private String getV3RoutingService(String [] uriParts, AllowedMethod requestMeth
         return Constants.Service.STREAMS;
       }
     } else if ((uriParts.length >= 8 && uriParts[7].equals("logs")) ||
+      (uriParts.length >= 10 && uriParts[9].equals("logs")) ||
```

Comment:
Some URLs cannot be prefix-matched. I think we need to do routing by getting all the URLs that need to be routed, and pattern match on it.